### PR TITLE
idobo/drewdobo keymap v1

### PR DESCRIPTION
**Pull request by djfroofy**
##  Description

This change adds a new keymap for idobo keyboard which matches my preferences: space, enter, backspace and delete controlled with thumbs. 

## Types of Changes

- [x] Keymap/layout/userspace (addition)

### CODE REVIEW

File: keyboards/idobo/keymaps/drewdobo/keymap.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+// Keyboard Layers
+#define _QW 0
```
Comment (noroadsleft) [inline keyboards/idobo/keymaps/drewdobo/keymap.c]: I'd recommend using an enum to manage your layers here in place of Lines 20-23:

```c

enum layer_names {
    _QW,
    _QW_MAC,
    _FN,
    _MOUSE,
};
```

Doing it this way means adding/deleting/reordering only requires editing lines in the whole, as opposed to editing a bunch of hard definitions.

File: keyboards/idobo/keymaps/drewdobo/keymap.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#define _QW_MAC 1
+#define _FN 2
+#define _MOUSE 3
+
+// Some short-cut aliases
+#define Z_LALT LALT_T(KC_ESC)
+#define Z_MAC DF(_QW_MAC)
+#define Z_PC DF(_QW)
+#define ___N___ KC_NO
```
Comment (noroadsleft) [inline keyboards/idobo/keymaps/drewdobo/keymap.c]: QMK defines an alias for this keycode automatically for every keymap.

https://github.com/qmk/qmk_firmware/blob/140c08e521084410ccb239848a3253b13c12683d/quantum/quantum_keycodes.h#L31

File: keyboards/idobo/keymaps/drewdobo/keymap.c (hunk 1 above)
Comment (djfroofy) [reply keyboards/idobo/keymaps/drewdobo/keymap.c]: Ah nice, updated to use enum instead.

File: keyboards/idobo/keymaps/drewdobo/keymap.c (hunk 2 above)
Comment (djfroofy) [reply keyboards/idobo/keymaps/drewdobo/keymap.c]: Update to use alias already defined in QMK.

**Pull request merged**